test/AutoRest.TestServer.Tests/Mgmt/TestProjects/TestProjectTests.cs

```diff
@@ -157,8 +158,11 @@ private static void VerifyMethodReturnType(Type type, Type expectedType, MethodI
             Assert.AreEqual(expectedType, genericArgument, $"Return type did not match for {type.Name}.{method.Name}");
         }
 
-        private Type? GetResourceFromCollection(Type collection) => MyTypes().FirstOrDefault(t => t.Name == GetResourceNameFromCollectionName(collection.Name));
-        private string GetResourceNameFromCollectionName(string collectionName) => collectionName.Substring(0, collectionName.IndexOf("Collection"));
+        private Type? GetResourceFromCollection(Type collection)
+        {
+            var baseName = collection.Name.Substring(0, collection.Name.IndexOf("Collection"));
+            return MyTypes().FirstOrDefault(t => t.Name == baseName.AddResourceSuffixToResourceName() || t.Name == baseName);
```

Comment:
Since we have `no-resource-suffix` directive, here we cannot simply get the resource name by trimming `Collection` from the collection name.